### PR TITLE
Prover: fix credx credential retrieval

### PR DESCRIPTION
**Pull request by Patrik-Stas**
- Fixes retrieval of credentials for prover. When multiple restriction objects are specified, they should be treated as `OR` instead of `AND`. Verifier is asking prover to satisfy at least 1 restriction object, not all of them.

- Adds test `test_agency_pool_it_should_select_credentials_for_satisfiable_restriction` which was failing without modifications to WQL made in the PR. Passes with the modifications. See anoncreds spec https://hyperledger.github.io/anoncreds-spec/#restrictions

> A boolean expression is formed by ORing and ANDing the source credential properties. The following JSON is an example. Any of the source credential properties listed above can be used in the expression components:
> 
>       "restrictions": [
>         {
>           "issuer_did": "<did>",
>           "schema_id": "id"
>         },
>         {
>           "cred_def_id" : "<id>",
>           "attr::color::marker": "1",
>           "attr::color::value" : "red"
>         }
>       ]
> The properties in each list item are AND’d together, and the array elements are OR’d together. As such, the example above defines the logical expression:
> 
> The attributes must come from a source verifiable credential such that:
>    issuer_did = <did> AND
>      schema_id = <id>
>    OR
>    cred_def_id = <id> AND
>       the credential must contain an attribute name "color" AND
>       the credential must contain an attribute name "color" with the attribute value "red"

- Test refactoring also uncovered that test `test_agency_pool_it_should_fail_to_select_credentials_for_predicate` is not working with credx implementation; see: https://github.com/hyperledger/aries-vcx/issues/962

### CODE REVIEW

File: aries_vcx_core/src/anoncreds/credx_anoncreds.rs [hunk 1]
```diff
@@ -160,12 +160,13 @@ impl IndyCredxAnonCreds {
...
+                    attrs.push(restrictions_wql);
+                    json!({ "$and": attrs })
                 }
-                Value::Object(obj) => {
-                    attrs.push(Value::Object(obj));
+                Value::Object(restriction) => {
+                    attrs.push(Value::Object(restriction));
                     json!({ "$and": attrs })
                 }
                 _ => json!(attrs),
```
Comment (gmulhearn) [inline aries_vcx_core/src/anoncreds/credx_anoncreds.rs:172]: I suppose this case is us saying "ignore the restrictions if they aren't a valid restriction json type (obj or array)".

I wonder more generally if we should just throw an error here; "unsupported restriction". 

If not, I'm not sure json!(attrs) is correct for ignoring restrictions.

I think it should be json!({$and: attrs})

To indicate that ALL attr in the attrs array must be satisfied

Comment (gmulhearn) [reply aries_vcx_core/src/anoncreds/credx_anoncreds.rs:172]: Also a couple lines below (175) has the same issue

**Review by gmulhearn — CHANGES_REQUESTED** [review]
LGTM; but would be good to fix up my comment while you're in this area.

Good to merge after that

File: aries_vcx_core/src/anoncreds/credx_anoncreds.rs (hunk 1 above)
Comment (Patrik-Stas) [reply aries_vcx_core/src/anoncreds/credx_anoncreds.rs]: should be addressed, please check

Comment (gmulhearn) [reply aries_vcx_core/src/anoncreds/credx_anoncreds.rs]: Sorry for delay, your fox matches my understanding, thanks

**Review by gmulhearn — APPROVED** [review]
Fix from my last comment looks good

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -336,7 +336,6 @@ jobs:
         run: |
           cargo test --manifest-path="wallet_migrator/Cargo.toml";
           RUST_TEST_THREADS=1 CARGO_INCREMENTAL=0 TEST_POOL_IP=127.0.0.1 cargo test --manifest-path="aries_vcx/Cargo.toml" -F migration --test test_creds_proofs -- --include-ignored;
-          RUST_TEST_THREADS=1 CARGO_INCREMENTAL=0 TEST_POOL_IP=127.0.0.1 cargo test --manifest-path="aries_vcx/Cargo.toml" -F migration --test test_creds_proofs_revocations -- --include-ignored;
```
Comment (mirgee) [inline .github/workflows/main.yml:339]: I am sorry, why was this removed?

**Review by mirgee — APPROVED** [review]
